### PR TITLE
docs: Document ParentClosePolicy and ChildWorkflowCancellationType

### PR DESCRIPTION
**Pull request by lorensr**
Spencer checked `ChildWorkflowCancellationType`

### CODE REVIEW

File: packages/internal-workflow-common/src/workflow-options.ts [hunk 1]
```diff
@@ -20,10 +45,10 @@ export interface BaseWorkflowOptions {
    * Specifies server behavior if a completed workflow with the same id exists. Note that under no
    * conditions Temporal allows two workflows with the same namespace and workflow id run
    * simultaneously.
-   *   ALLOW_DUPLICATE_FAILED_ONLY is a default value. It means that workflow can start if
-   *   previous run failed or was canceled or terminated.
+   *   ALLOW_DUPLICATE_FAILED_ONLY is a default value. It means that
```
Comment (bergundy) [inline packages/internal-workflow-common/src/workflow-options.ts:48]: What was the intent here?

File: packages/workflow/src/interfaces.ts [hunk 2]
```diff
@@ -165,19 +165,68 @@ export interface ContinueAsNewOptions {
...
+ * Specifies:
+ * - whether cancellation requests are sent to the Child
+ * - whether and when a {@link CanceledFailure} is thrown from {@link executeChild} or
+ *   {@link ChildWorkflowHandle.result}
+ *
+ * @default {@link ChildWorkflowCancellationType.WAIT_CANCELLATION_COMPLETED}
+ */
 export enum ChildWorkflowCancellationType {
+  /**
+   * Don't send a cancellation request to the Child. Don't throw an error.
```
Comment (bergundy) [inline packages/workflow/src/interfaces.ts:178]: Not sure this is right, I think that the error does get thrown but I'd need to verify.

I don't think we test all of these behaviors, I guess we will in the sdk-features repo.

File: packages/workflow/src/interfaces.ts [hunk 3]
```diff
@@ -165,19 +165,68 @@ export interface ContinueAsNewOptions {
...
+   */
   ABANDON = 0,
+
+  /**
+   * Send a cancellation request to the Child. Immediately throw the error.
+   */
   TRY_CANCEL = 1,
+
+  /**
+   * Send a cancellation request to the Child. Throw the error if and when the Child's cancellation is completed.
```
Comment (bergundy) [inline packages/workflow/src/interfaces.ts:188]: I'd note that the child might not cooperate and might end up in either failed / cancelled or completed states (or never complete).

File: packages/workflow/src/interfaces.ts [hunk 4]
```diff
@@ -165,19 +165,68 @@ export interface ContinueAsNewOptions {
...
+/**
+ * Concept: [Parent Close Policy](https://docs.temporal.io/concepts/what-is-a-parent-close-policy/)
+ *
+ * How a Child Workflow reacts to the Parent Workflow reaching a Closed state.
+ */
 export enum ParentClosePolicy {
+  /**
+   * No need to use this.
+   *
+   * (If a `ParentClosePolicy` is set to this, or is not set at all, the default value will be used.)
```
Comment (bergundy) [inline packages/workflow/src/interfaces.ts:211]: I'd just go with:
```suggestion
   * If a `ParentClosePolicy` is set to this, or is not set at all, the server default value will be used.
```

File: packages/internal-workflow-common/src/workflow-options.ts (hunk 1 above)
Comment (lorensr) [reply packages/internal-workflow-common/src/workflow-options.ts:48]: Oops thanks—meant to move to the enum values

**Review by bergundy — APPROVED** [review]

**Pull request merged**